src/raw/serde/seeded_visitor.rs

```diff
@@ -272,122 +264,88 @@ impl<'a, 'de> Visitor<'de> for SeededVisitor<'a, 'de> {
+                            RawBsonRef::Symbol(s) => {
+                                self.append_string(s);
+                                Ok(ElementType::Symbol)
+                            }
+                            RawBsonRef::Decimal128(d) => {
+                                self.buffer.append_bytes(&d.bytes);
+                                Ok(ElementType::Decimal128)
+                            }
+                            RawBsonRef::RegularExpression(re) => {
+                                self.append_cstring(re.pattern)
+                                    .map_err(SerdeError::custom)?;
+                                self.append_cstring(re.options)
+                                    .map_err(SerdeError::custom)?;
+                                Ok(ElementType::RegularExpression)
+                            }
+                            RawBsonRef::Undefined => Ok(ElementType::Undefined),
+                            RawBsonRef::DateTime(dt) => {
+                                self.buffer.append_bytes(&dt.to_le_bytes());
+                                Ok(ElementType::DateTime)
+                            }
+                            RawBsonRef::Timestamp(ts) => {
+                                self.buffer.append_bytes(&ts.increment.to_le_bytes());
+                                self.buffer.append_bytes(&ts.time.to_le_bytes());
+                                Ok(ElementType::Timestamp)
+                            }
+                            RawBsonRef::MinKey => Ok(ElementType::MinKey),
+                            RawBsonRef::MaxKey => Ok(ElementType::MaxKey),
+                            RawBsonRef::JavaScriptCode(s) => {
+                                self.append_string(s);
+                                Ok(ElementType::JavaScriptCode)
+                            }
+                            RawBsonRef::JavaScriptCodeWithScope(jsc) => {
+                                let length_index = self.pad_document_length();
+                                self.append_string(jsc.code);
+                                self.buffer.append_bytes(jsc.scope.as_bytes());
+
+                                let length_bytes =
+                                    ((self.buffer.len() - length_index) as i32).to_le_bytes();
+                                self.buffer
+                                    .copy_from_slice(length_index..length_index + 4, &length_bytes);
+
+                                Ok(ElementType::JavaScriptCodeWithScope)
+                            }
+                            RawBsonRef::DbPointer(dbp) => {
+                                self.append_string(dbp.namespace);
+                                self.buffer.append_bytes(&dbp.id.bytes());
+                                Ok(ElementType::DbPointer)
+                            }
+                            // No other `Leaf` arms are emitted by `parse_map`
+                            _ => unreachable!(),
```

Comment:
I considered introducing a new enum so this could be exhaustive, but I believe this is the only place where case-by-case matching of this sort will be needed so it didn't seem worth it.